docs/docsite/rst/dev_guide/developing_inventory.rst

```diff
@@ -1,21 +1,220 @@
+
+
+
+.. _developing_inventory_inventory_plugins:
+
+Inventory Plugins
+-----------------
+
+Like most plugin types (except modules) they must be developed in Python, since they execute on the controller they should match the same requirements :ref:`control_machine_requirements`.
+
+Most of the documentation in :ref:`developing_plugins` also applies here, so as to not repeat ourselves, you should read that document first and we'll include inventory plugin specifics next.
+
+Inventory plugins normally only execute at the start of a run, before playbooks/plays and roles are loaded,
+but they can be 're-executed' via the ``meta: refresh_inventory`` task, which will clear out the existing inventory and rebuild it.
+
+When using the 'persistent' cache, inventory plugins can also use the configured cache plugin to store and retrieve data to avoid costly external calls.
+
+.. _developing_an_inventory_plugin:
+
+Developing an Inventory Plugin
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first thing you want to do is use the base class:
+
+.. code-block:: python
+
+    from ansible.plugins.inventory import BaseInventoryPlugin
+
+    class InventoryModule(BaseInventoryPlugin):
+
+        NAME = 'myplugin'  # used internally by Ansible, it should match the file name but not required
+
+This class has a couple of methods each plugin should implement and a few helpers for parsing the inventory source and updating the inventory.
+
+After you have the basic plugin working you might want to to incorporate other features by adding more base classes:
+
+.. code-block:: python
+
+    from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+    class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+        NAME = 'myplugin'
+
+For the bulk of the work in the plugin, We mostly want to deal with 2 methods ``verify_file`` and ``parse``.
+
+.. _inventory_plugin_verify_file:
+
+verify_file
+"""""""""""
+
+This method is used by Ansible to make a quick determination if the inventory source is usable by the plugin. It does not need to be 100% accurate as there might be overlap in what plugins can handle and Ansible will try the enabled plugins (in order) by default.
+
+.. code-block:: python
+
+    def verify_file(self, path):
+        ''' return true/false if this is possibly a valid file for this plugin to consume '''
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            # base class verifies that file exists and is readable by current user
+            if path.endswith(('.vbox.yaml', '.vbox.yml')):
+                valid = True
+        return valid
+
+In this case, from the :ref:`virtualbox inventory plugin <virtualbox_inventory>`, we screen for specific file name patterns to avoid attempting to consume any valid yaml file. You can add any type of condition here, but the most common ones are 'extension matching'
```

Comment:
Did you mean to add another item here? " . . . the most common ones are `extension matching`" . . . and???

Reply:
should be 'most common one'